openff/bespokefit/schema/targets.py

```diff
@@ -20,6 +23,130 @@
 from openff.bespokefit.utilities.pydantic import SchemaBase
 
 
+def _check_connectivity(
+    qcschema: QCEMolecule,
+    name: str,
+    fragment: Optional[Molecule] = None,
+):
+    """
+    Raise an exception if the geometry of ``qcschema`` does not match ``fragment``
+
+    Parameters
+    ==========
+
+    qcschema
+        A QCElemental ``Molecule`` representing the final geometry of a QC
+        computation
+    name
+        A name for the current calculation. Used in the exception raised by this
+        method.
+    fragment
+        An OpenFF ``Molecule`` representing the true chemical identity of the
+        fragment.
+
+    Returns
+    =======
+
+    None
+
+    Raises
+    ======
+
+    ValueError
+        If the connectivity does not match.
+    """
+    # If expected connectivity is not provided, compute it from the fragment
+    if fragment is None:
+        fragment = Molecule.from_qcschema(qcschema)
+
+    # Get expected connectivity from bonds
+    expected_connectivity = {
```

Comment:
I think the expected connectivity should be in the qcschema molecule which would save calculating each time.

Reply:
I think that's true for `LocalQCData` but not for `TorsionDriveRecord` etc